MapboxNavigation/NavigationMapView.swift

```diff
@@ -427,36 +426,82 @@ open class NavigationMapView: MGLMapView {
             }
         }
     }
-    
     /**
-     Fired when the user taps a route. If there is an alternate route, and an alternate route is tapped, it is selected.
+     Fired when NavigationMapView detects a tap not handled elsewhere by other gesture recognizers.
      */
-    func didTapRoute(tap: UITapGestureRecognizer) {
-        guard let routes = routes else { return }
-        
-        let tapPoint = tap.location(in: self)
-        let tapCoordinate = self.convert(tapPoint, toCoordinateFrom: tap.view)
-        
-        let closestRoute = routes.filter {
-            guard let coords = $0.coordinates else { return false }
-            return coords.count > 1
-        }.min { (left, right) -> Bool in
-            let leftDistance = Polyline(left.coordinates!).closestCoordinate(to: tapCoordinate)!.distance
-            let rightDistance = Polyline(right.coordinates!).closestCoordinate(to: tapCoordinate)!.distance
+    func didRecieveTap(sender: UITapGestureRecognizer) {
+        guard let routes = routes, let tapPoint = sender.point else { return }
+        
+        let waypointTest = waypoints(on: routes, closeTo: tapPoint) //are there waypoints near the tapped location?
```

Comment:
If there was a waypoint near the tapped location, its annotation would’ve gotten selected, so `mapView.selectedAnnotations` would be non-empty.

Reply:
Intermediate waypoints are map features, not `MGLAnnotation`s. https://github.com/mapbox/mapbox-navigation-ios/blob/master/MapboxNavigation/NavigationMapView.swift#L516